### PR TITLE
Merge `DocC` documentation target and `docc` executable target

### PR DESCRIPTION
**Pull request by ethan-kusters**
Bug/issue #, if applicable: 

## Summary

Now that Swift-DocC supports executable targets there are two documentable targets named “docc” when lowercased which is causing issues on case-insensitive file systems.

This resolves the issue by merging DocC’s conceptual documentation into the `docc` executable target. There should be no user-facing change here.

## Testing

Checkout this branch and run:

```
swift package --disable-sandbox preview-documentation --target docc
```

confirm documentation renders as expected.

Additionally run

```
swift package generate-documentation
```

and confirm that no documentation output is overwritten like it is when running this command on the current main branch.

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- [ ] ~~Added tests~~
- [x] Ran the `./bin/test` script and it succeeded
- [x] Updated documentation if necessary

### CODE REVIEW

**Comment by ethan-kusters** [comment]
@swift-ci please test

**Review by franklinsch — COMMENTED** [review]
Neat, these changes look great. Do we also need to update `bin/preview-docs`?

**Comment by ethan-kusters** [comment]
> Neat, these changes look great. Do we also need to update bin/preview-docs?

Good catch, thank you! Updated here: [5783e10](https://github.com/apple/swift-docc/pull/275/commits/5783e1029c2f87cce2210e118c2fe41f8a5d3ce3)

**Review by franklinsch — APPROVED** [review]

**Comment by Kyle-Ye** [comment]
Any progress on this? @ethan-kusters 

Could we rebase and merge it?

**Comment by ethan-kusters** [comment]
@swift-ci please test

File: Package.resolved [hunk 1]
```diff
@@ -24,8 +24,8 @@
         "repositoryURL": "https://github.com/apple/swift-crypto.git",
         "state": {
           "branch": null,
-          "revision": "ddb07e896a2a8af79512543b1c7eb9797f8898a5",
-          "version": "1.1.7"
+          "revision": "9680b7251cd2be22caaed8f1468bd9e8915a62fb",
+          "version": "1.1.2"
```
Comment (Kyle-Ye) [inline Package.resolved:28]: Maybe we should discard this before merge. 😄

**Review by Kyle-Ye — CHANGES_REQUESTED** [review]
There is some downgrade dependency on Package.resolved. Maybe we should discard the change in that file? 🤔

File: Package.resolved (hunk 1 above)
Comment (ethan-kusters) [reply Package.resolved]: Good catch- thanks!

**Comment by ethan-kusters** [comment]
@swift-ci please test

**Review by Kyle-Ye — APPROVED** [review]

**Comment by Kyle-Ye** [comment]
Could we squash and merge this? @ethan-kusters

**Pull request merged**